sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSource.scala

```diff
@@ -724,4 +726,35 @@ object DataSource extends Logging {
          """.stripMargin)
     }
   }
+
+  /**
+   * Return all paths represented by the wildcard string.
+   * Use a local thread pool to do this while there's too many paths.
+   */
+  private def getGlobbedPaths(
+      sparkSession: SparkSession,
+      fs: FileSystem,
+      hadoopConf: Configuration,
+      qualified: Path): Seq[Path] = {
+    val getGlobbedPathThreshold = sparkSession.sessionState.conf.parallelGetGlobbedPathThreshold
+    val paths = SparkHadoopUtil.get.expandGlobPath(fs, qualified, getGlobbedPathThreshold)
```

Comment:
If the code has any unexpected side effect, is that possible to turn it off and restore it to the previous code path?

Reply:
em...It's hard to achieve this in currently implement, maybe use a bool config like `spark.sql.sources.parallelGetGlobbedPath.enable` to control this and previous code path here? Maybe safer under your concerns.

Reply:
We could disable this feature by setting a negative value  into `spark.sql.sources.parallelGetGlobbedPath.numThreads` without adding a new config.

Reply:
Thanks for your advise, I'll reuse the value of `spark.sql.sources.parallelGetGlobbedPath.numThreads` to control this.